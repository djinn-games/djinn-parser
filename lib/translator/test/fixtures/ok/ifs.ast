{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "ifs",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "bool",
          "id": {
            "type": "Identifier",
            "name": "flag",
            "line": 3
          },
          "init": {
            "type": "Literal",
            "line": 3,
            "dataType": "bool",
            "value": true
          },
          "line": 3
        },
        "line": 3
      },
      {
        "type": "IfSentence",
        "if": {
          "condition": {
            "type": "Identifier",
            "name": "flag",
            "line": 4
          },
          "consequent": [
            {
              "type": "ExpressionSentence",
              "expression": {
                "type": "Identifier",
                "name": "flag",
                "line": 5
              },
              "line": 5
            },
            {
              "type": "ExpressionSentence",
              "expression": {
                "type": "VarDeclaration",
                "dataType": "bool",
                "id": {
                  "type": "Identifier",
                  "name": "flag",
                  "line": 6
                },
                "init": {
                  "type": "Literal",
                  "line": 6,
                  "dataType": "bool",
                  "value": false
                },
                "line": 6
              },
              "line": 6
            }
          ],
          "alternates": []
        }
      }
    ],
    "line": 2
  },
  "line": 1
}