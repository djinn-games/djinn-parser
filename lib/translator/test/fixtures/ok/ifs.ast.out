{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_flag",
            "raw": "\"_flag\""
          },
          "computed": true
        },
        "right": {
          "type": "Literal",
          "value": true,
          "raw": "true"
        }
      }
    },
    {
      "type": "IfStatement",
      "test": {
        "type": "MemberExpression",
        "object": {
          "type": "MemberExpression",
          "object": {
            "type": "Identifier",
            "name": "DJINN"
          },
          "property": {
            "type": "Identifier",
            "name": "__scope"
          }
        },
        "property": {
          "type": "Literal",
          "value": "_flag",
          "raw": "\"_flag\""
        },
        "computed": true
      },
      "consequent": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ExpressionStatement",
            "expression": {
              "type": "MemberExpression",
              "object": {
                "type": "MemberExpression",
                "object": {
                  "type": "Identifier",
                  "name": "DJINN"
                },
                "property": {
                  "type": "Identifier",
                  "name": "__scope"
                }
              },
              "property": {
                "type": "Literal",
                "value": "_flag",
                "raw": "\"_flag\""
              },
              "computed": true
            }
          },
          {
            "type": "ExpressionStatement",
            "expression": {
              "type": "AssignmentExpression",
              "operator": "=",
              "left": {
                "type": "MemberExpression",
                "object": {
                  "type": "MemberExpression",
                  "object": {
                    "type": "Identifier",
                    "name": "DJINN"
                  },
                  "property": {
                    "type": "Identifier",
                    "name": "__scope"
                  }
                },
                "property": {
                  "type": "Literal",
                  "value": "__flag",
                  "raw": "\"__flag\""
                },
                "computed": true
              },
              "right": {
                "type": "Literal",
                "value": false,
                "raw": "false"
              }
            }
          }
        ]
      },
      "alternate": {
        "type": "BlockStatement",
        "body": []
      }
    }
  ]
}
